syntax = "proto3";

option java_multiple_files = true;
option java_package = "io.grpc.examples.directory";
option java_outer_classname = "DirectoryProto";
option objc_class_prefix = "DIR";

package directory;

service Client {
    rpc pageWrite (PageRequest) returns (PageReply) {}

    rpc pageRead (PageRequest) returns (PageReply) {}
}

service Master {
    rpc fwdPageRequest (PageRequest) returns (PageReply) {}

    rpc invPage (PageRequest) returns (PageReply) {}
}

message PageRequest {
    uint64 pageAddr= 1;
    uint32 pageSize = 2;
    uint32 pageOperation = 3; // 0 for Read, 1 for Write, 2 for Invalidate
}

message PageReply {
    bool ack = 1; // True for ack, False for nack
    bytes pageData = 2;
}
